libcnb/src/lib.rs

```diff
@@ -78,6 +82,39 @@ macro_rules! buildpack_main {
     };
 }
 
+/// Resolves the path to an additional buildpack binary by Cargo target name when the buildpack is
+/// packaged with `libcnb-cargo` or `libcnb-test`.
+///
+/// This can be used to copy additional binaries to layers or use them for exec.d.
+///
+/// To add an additional binary to a buildpack, add a new file with a main function to `bin/`.
+/// Cargo will [automatically configure it as a binary target](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#target-auto-discovery)
+/// with the name of file.
+///
+/// Note: This only works properly if the buildpack is packaged with `libcnb-cargo`/`libcnb-test`.
+#[macro_export]
+macro_rules! additional_buildpack_binary_path {
```

Comment:
I presume it's not possible to add unit tests for this macro since:
- the failing case won't compile (so can't be tested without having some kind of runner that invokes Rust)
- the passing case won't work in this crate, since there are no binary targets we can test this with?

Also, even if we could test the passing case at least compiles, it wouldn't be testing the runtime aspect anyway.

As such, testing this only via a future integration test (eg the new examples-03 I see drafted in #311) seems fine to me :-)